src/dbos.ts

```diff
@@ -902,11 +909,17 @@ export class DBOS {
         else {
           inst = this as ConfiguredInstance;
           if (!("name" in inst)) {
-            throw new DBOSInvalidWorkflowTransitionError();
+            throw new DBOSInvalidWorkflowTransitionError("Attempt to call a `step` function on an object that is not a `ConfiguredInstance`");
           }
         }
 
         if (DBOS.isWithinWorkflow()) {
+          if (DBOS.isInTransaction()) {
+            throw new DBOSInvalidWorkflowTransitionError("Invalid call to a `step` function from within a `transaction`");
+          }
+          if (DBOS.isInStep()) {
+            throw new DBOSInvalidWorkflowTransitionError("Invalid call to a `step` function from within a `step`");
```

Comment:
So what's the correct way to call another step function without triggering the decorator? Previously, we can dierctly call the function without using `invoke` and by passing the same context. Maybe we should just allow such embedded calls? (Could be a separate PR)

Reply:
This is a good question (and for transactions also).  I wonder what Python is currently doing, so that was the next step - write the same test in Python if we don't have it already.

I'm OK with relaxing the rule and calling the function directly, but currently you'd make a shared core function.

Reply:
Here's how we handle it in Python: https://github.com/dbos-inc/dbos-transact-py/blob/main/dbos/_core.py#L702-L704
```python
# Entering step is allowed:
#  In a step already, just call the original function directly.
#  In a workflow (that is not in a step already)
#  Not in a workflow (we will start the single op workflow)
ctx = get_local_dbos_context()
if ctx and ctx.is_step():
    # Call the original function directly
    return func(*args, **kwargs)
```

Reply:
I think if it's easy to do, we should implement the same logic in TypeScript

Reply:
Added that logic.
I do wonder if we should be checking for agreement between the config options at the transition point... but we can do it later.